server/routes/stripe.js

```diff
@@ -103,4 +104,28 @@ router.post('/webhook', bodyParser.raw({ type: 'application/json' }), async (req
   res.json({ received: true });
 });
 
+// called from frontend after stripe payment confirmation
+// redirect to a thank you page
+router.get('/payment/success/:id', redirectLogin, async (req, res) => {
+  try {
```

Comment:
you should make sure that only the user that donated can see this page, otherwise someone could farm all of the data by just opening this page with a wishcardid

Reply:
I cant make that. Since the donation hook is inserting data into the donation and this is called immediately after payment confirmation. Plus all of the data that is visible here is visible on wishcard itself which is public